src/tests/dbus-tests/test_40_drive.py

```diff
@@ -0,0 +1,118 @@
+        new_dirs = list(set(new_dirs) - set(dirs))
+        assert len(new_dirs) == 1
+
+        self.cd_dev = os.listdir(new_dirs[0])
+        assert len(self.cd_dev) == 1
+        self.cd_dev = '/dev/' + self.cd_dev[0]
+        assert os.path.exists(self.cd_dev)
+
+        self.cd_device = self.get_device(self.cd_dev)
+        self.cd_drive_name = self.get_drive_name(self.cd_device)
+        self.cd_drive = self.get_drive(self.cd_device)
+
+
+    def tearDown(self):
+        device = self.cd_dev.split('/')[-1]
+        if os.path.exists('/sys/block/' + device):
+            with open('/sys/block/%s/device/delete' % device, 'w') as f:
+                f.write('1')
+            while os.path.exists(device):
+                time.sleep(0.1)
+            subprocess.call(['udevadm', 'settle'])
+            subprocess.call(['modprobe', '-r', 'scsi_debug'])
+
+
+    def test_10_eject(self):
+        ''' Test of Drive.Eject method '''
+        # when new ejectable device is added using scsi_debug,
+        # it affects already existing virtual devices making some of them
+        # ejectable as well; this is a workaround
+
+        for d in self.vdevs + [self.cd_dev]:
+            dev = self.get_device(d)
+            drive = self.get_drive(dev)
+            drive_name = self.get_drive_name(dev)
+            if drive_name == self.cd_drive_name:
+                # ejectable devices should support the Eject function
+                try:
+                    drive.Eject(self.no_options)
+                except Exception:
+                    self.fail("An exception occured while calling Eject function")
+            else:
+                # not ejectable devices should try to run bash command and fail
+                # with specific exception description
+                with self.assertRaisesRegex(dbus.exceptions.DBusException,
+                        'is not hot-pluggable device'):
+                    drive.Eject(self.no_options)
+
+
+    def test_20_poweroff(self):
+        ''' Test of Drive.PowerOff method '''
+        for d in self.vdevs + [self.cd_dev]:
+            device = self.get_device(d)
+            drive =  self.get_drive(device)
+
+            # check should fail since device cannot be powered off
+            # sadly so far the only way we can test this function
+            with self.assertRaisesRegex(dbus.exceptions.DBusException,
+                    'Failed: No usb device'):
+                drive.PowerOff(self.no_options)
+
+
+    def test_30_setconfiguration(self):
+        ''' Test of Drive.SetConfiguration method '''
+        # set configuration value to some improbable value
+        self.cd_drive.SetConfiguration({'ata-pm-standby': 286}, self.no_options)
+
+        # validate that configration property has changed
+        conf_value = self.get_property(self.cd_drive, '.Drive', 'Configuration')
+        self.assertEqual(str(conf_value['ata-pm-standby']), '286')
```

Comment:
`Drive` interface has a lot of properties, we should test at least some of them (compare dbus properties with some "system" values accessible using `lsblk` etc.)